src/control/cmd/dmg/pool.go

```diff
@@ -72,68 +65,101 @@ type PoolCreateCmd struct {
+	ScmRatio   float64 `short:"t" long:"scm-ratio" default:"6" description:"Percentage of SCM:NVMe for pool storage (auto)"`
+	NumRanks   uint32  `short:"k" long:"nranks" description:"Number of ranks to use (auto)"`
+	NumSvcReps uint32  `short:"v" long:"nsvc" description:"Number of pool service replicas"`
+	ScmSize    string  `short:"s" long:"scm-size" description:"Per-server SCM allocation for DAOS pool (manual)"`
+	NVMeSize   string  `short:"n" long:"nvme-size" description:"Per-server NVMe allocation for DAOS pool (manual)"`
+	RankList   string  `short:"r" long:"ranks" description:"Storage server unique identifiers (ranks) for DAOS pool"`
+	Sys        string  `short:"S" long:"sys" default:"daos_server" description:"DAOS system that pool is to be a part of"`
 }
 
 // Execute is run when PoolCreateCmd subcommand is activated
 func (cmd *PoolCreateCmd) Execute(args []string) error {
-	msg := "SUCCEEDED: "
-
-	scmBytes, err := humanize.ParseBytes(cmd.ScmSize)
-	if err != nil {
-		return errors.Wrap(err, "pool SCM size")
+	if cmd.Size != "" && (cmd.ScmSize != "" || cmd.NVMeSize != "") {
+		return errIncompatFlags("size", "scm-size", "nvme-size")
 	}
-
-	var nvmeBytes uint64
-	if cmd.NVMeSize != "" {
-		nvmeBytes, err = humanize.ParseBytes(cmd.NVMeSize)
-		if err != nil {
-			return errors.Wrap(err, "pool NVMe size")
-		}
+	if cmd.Size == "" && cmd.ScmSize == "" {
+		return errors.New("either --size or --scm-size must be supplied")
 	}
 
-	ratio := 1.00
-	if nvmeBytes > 0 {
-		ratio = float64(scmBytes) / float64(nvmeBytes)
+	calcRatio := func(scm, nvme uint64) float64 {
+		if nvme == 0 {
+			return 1.00
+		}
+		return float64(scm) / float64(nvme)
 	}
 
-	if ratio < minScmNvmeRatio {
-		cmd.log.Infof("SCM:NVMe ratio is less than %0.2f %%, DAOS "+
-			"performance will suffer!\n", ratio*100)
+	var err error
+	req := &control.PoolCreateReq{
+		User:       cmd.UserName,
+		UserGroup:  cmd.GroupName,
+		Sys:        cmd.Sys,
+		Name:       cmd.PoolName,
+		NumSvcReps: cmd.NumSvcReps,
 	}
-	cmd.log.Infof("Creating DAOS pool with %s SCM and %s NVMe storage "+
-		"(%0.2f %% ratio)\n", humanize.Bytes(scmBytes),
-		humanize.Bytes(nvmeBytes), ratio*100)
 
-	var acl *control.AccessControlList
 	if cmd.ACLFile != "" {
-		acl, err = control.ReadACLFile(cmd.ACLFile)
+		req.ACL, err = control.ReadACLFile(cmd.ACLFile)
 		if err != nil {
 			return err
 		}
 	}
 
-	if cmd.NumSvcReps > maxNumSvcReps {
-		return errors.Errorf("max number of service replicas is %d, got %d",
-			maxNumSvcReps, cmd.NumSvcReps)
-	}
-
-	ranks, err := system.ParseRanks(cmd.RankList)
+	req.Ranks, err = system.ParseRanks(cmd.RankList)
 	if err != nil {
 		return errors.Wrap(err, "parsing rank list")
 	}
 
-	req := &control.PoolCreateReq{
-		ScmBytes: scmBytes, NvmeBytes: nvmeBytes, Ranks: ranks,
-		NumSvcReps: cmd.NumSvcReps, Sys: cmd.Sys,
-		User: cmd.UserName, UserGroup: cmd.GroupName, ACL: acl,
+	if cmd.Size != "" {
```

Comment:
Just a suggestion, but for improved clarity I think it'd be nice to break out the blocks into helper functions (updateReqStorageAuto, updateReqStorageManual, or something like that).

Reply:
Agreed that this could be cleaner. Longer-term, I am thinking that the manual options may go away though. I don't really see valid use cases for them other than compatibility with older scripts.